### ISSUE TITLE
LongMethod should not consider parameters while calculating the number of lines

### ISSUE THREAD

**Issue by niraj8** [detekt/detekt#2804]
## Expected Behavior

LongMethod should consider the method definition without the input parameters.
Example:
If `LongMethod` threshold is set to 7, then the below method shouldn't be tagged as the method implementation is actually 3 loc and detekt already has `LongParameterList` to check the param list
```
fun fetch(
        user: User,
        org: Org,
        @QueryValue startDate: String,
        @QueryValue endDate: String
    ) {
        val data = service1.get(startDate, endDate, context.user, context.org)
        val moreData = service2.get(startDate)
        ok(mapOf("data" to data+moreData))
    }
```
## Observed Behavior

The above snippet of code gets tagged with `LongMethod` code smell

## Steps to Reproduce

#2806

## Context

More LongMethod code smell then expected

## Your Environment

* Version of detekt used:1.9.1
* Version of Gradle used (if applicable):6.1.1
* Operating System and version:macOS 10.15.5
* Link to your project (if it's a public repository):

**Comment by cortinico** [comment]
Agree, this sounds like a false positive. Would you be able to work on a fix and make the tests in #2806 green (thank you for doing so btw)? We can support you in that if needed 👍

**Comment by niraj8** [comment]
Yes, I am working on it👍

**Comment by cortinico** [comment]
Assigning the issue to you so that we don't step on your toes 👍

**Comment by niraj8** [comment]
I wanted to get opinions from others before making any further changes.

For a top-level method, should the lines used by a nested method parameter list be considered in calculating the method length.

In the example below, the top-level body has 4 lines if we ignore the nested method.
```
fun longMethod(
    param1: String
) { // 4 lines
    println()
    println()

    fun nestedLongMethod(
        param1: String
    ) { // 5 lines
        println()
        println()
        println()
    }
}
```

Looks good?

**Comment by arturbosch** [comment]
It is a good practice to decompose a large function into smaller ones. In kotlin we have local functions. So we should exclude them completely (whole signature).

I've answered in #2806; so basically  #loc of longMethod.body - #loc of whole nestedLongMethod
